tests/xtimer_usleep_short/main.c

```diff
@@ -22,18 +22,54 @@
 #define TEST_USLEEP_MIN (0)
 #define TEST_USLEEP_MAX (500)
 
+#ifdef BOARD_NATIVE
+/* native can sometime take more time to respond as it is not real time */
+#define TEST_XTIMER_USLEEP_SHORT_SLEEP_MARGIN_US (1000)
+#else
+#define TEST_XTIMER_USLEEP_SHORT_SLEEP_MARGIN_US (20)
+#endif /* BOARD_NATIVE */
+
 int main(void)
 {
     xtimer_sleep(3);
     printf("This test will call xtimer_usleep for values from %d down to %d\n",
-            TEST_USLEEP_MAX, TEST_USLEEP_MIN);
+           TEST_USLEEP_MAX, TEST_USLEEP_MIN);
+
+    printf("Expected delay margin is %d us\n",
+           TEST_XTIMER_USLEEP_SHORT_SLEEP_MARGIN_US);
+
+    uint32_t test_time = 0, sleeping_time = 0, margin_fault = 0;
 
     for (int i = TEST_USLEEP_MAX; i >= TEST_USLEEP_MIN; i--) {
-        printf("going to sleep %d usecs...\n", i);
+        printf("going to sleep %d us\n", i);
+        uint32_t start = xtimer_now_usec();
         xtimer_usleep(i);
+        uint32_t slept = xtimer_now_usec() - start;
```

Comment:
making two `xtimer_now` calls is not necessary when saving the last (from previous iteration) and also assuming that all the other calls cost only marginal time compared to the sleep. `xtimer_now` is rather expensive, anyways.

Reply:
The print of  `"going to sleep %d us\n"`, if i am not mistaken, needs:
20 * 10 * 1000/9600Buad = 20,8 ms

Then the second print will also need some time.
This test goes from 500ms down to 1ms. So is the time marginal or not?

Reply:
I am also in favor of doing two times `xtimer_now` it helped timing tests correctly, printf takes time.